<?xml version="1.0"?>
<launch>

  <arg name="verbose_publishing" default="true"/> <!-- If false the node will only publish when there is some face detected -->
  <arg name="rate"               default="5"/>    <!-- Rate at which the node will look for faces and publish -->
  <arg name="image_topic"        default="/xtion/rgb/image_raw"/>

  <node name="pal_face" pkg="pal_face_detector_opencv" type="pal_face_detector_opencv" args="$(arg verbose_publishing) $(arg rate)" output="screen">
    <remap from="/pal_face/image" to="$(arg image_topic)"/>
    <rosparam>
      processing_img_width:  320
      processing_img_height: 240
      rel_min_eye_dist:      0.02  
      rel_max_eye_dist:      0.2
    </rosparam>
  </node>

  <node name="head_action" pkg="pal_face_detector_opencv" type="head_action" /> <!-- -->

  <node name="image_view_face" pkg="image_view" type="image_view">
    <remap from="image" to="pal_face/debug" />
    <remap from="image_transport" to="compressed" />
  </node>
</launch>
